sunpy/net/helio/parser.py

```diff
@@ -218,18 +218,11 @@ def wsdl_retriever(service='HEC'):
     def fail():
         raise ValueError("No online HELIO servers can be found.")
     service_links = webservice_parser(service=service)
-    wsdl = None
-    wsdl_links = None
-    if service_links is None:
-        fail()
-    for link in service_links:
-        wsdl_links = taverna_parser(link)
-    if wsdl_links is None:
-        fail()
-    for end_point in wsdl_links:
-        if end_point is not None and link_test(end_point) is not None:
-            wsdl = end_point
-            break
-    if wsdl is None:
-        fail()
-    return wsdl
+    if service_links is not None:
+        for link in service_links:
+            wsdl_links = taverna_parser(link)
+            if wsdl_links is not None:
+                for end_point in wsdl_links:
+                    if end_point is not None and link_test(end_point) is not None:
+                        return end_point
```

Comment:
too many indentations... that smells like we are doing something wrong (we may not, though...)
Probably we need to change how `taverna_parser` works.

Reply:
yeah, though I have removed `is not None` so now it looks bit clean.
However the indendations seems necessary. What we are doing is, checking if the object exists and iterating over its subobjects, and doing this again till we get `end_point`.
 
If it doesn't happen even one time for whole loops, we return `fail()`. Logic seems correct to me.